Adds spray-on insuls.

yeeyeh opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

Major props to everyone in #coding-general on the discord for putting up with my near complete inability to use .dm

## About The Pull Request
Adds a minor tech to the B.E.P.I.S.: spray-on insulated gloves. They come in a one-use can that can either be used on yourself or on an unsuspecting player to instantly apply a pair of insulated gloves. These gloves cannot be removed by standard means, as they are stuck to your skin. The only way to "safely" get them off is to degrade them via electric shock.
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
As of writing this, the B.E.P.I.S. has 3 major rewards and 3 minor rewards. I feel that it needs more minor rewards than major rewards to reinforce the fact that they are, in fact, minor.
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl:yeeye
add: Nanotrasen's research division's recent science fair has given light to an amazing, electrically-insulated compound that can be sprayed directly onto the skin. The initial idea was for footwear, but we all know what you're really going to do with it. Research can be continued at your station's B.E.P.I.S. unit.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

yeeyeh committed:
Adds spray-on insuls

necromanceranne commented:
I hate it and love it at the same time.

LemonInTheDark commented on code/modules/research/bepis.dm:
```diff
@@ -133,21 +133,23 @@
 		return
 	//Minor Failure
-	error_cause = pick("attempted to sell grey products to American dominated market.","attempted to sell gray products to British dominated market.","placed wild assumption that PDAs would go out of style.","simulated product #76 damaged brand reputation mortally.","simulated business model resembled 'pyramid scheme' by 98.7%.","product accidently granted override access to all station doors.")
+	error_cause = pick("attempted to sell grey products to American dominated market.","attempted to sell gray products to British dominated market.","placed wild assumption that PDAs would go out of style.","simulated product #76 damaged brand reputation mortally.","simulated business model resembled 'pyramid scheme' by 98.7%.","product accidentally granted override access to all station doors.","a ...
```
I'd say doc your changes but it's a meme so w/e

LemonInTheDark commented on code/modules/power/power.dm:
```diff
@@ -291,7 +291,7 @@
 		if(H.gloves)
 			var/obj/item/clothing/gloves/G = H.gloves
-			if(G.siemens_coefficient == 0)
+			if(G.siemens_coefficient == 0 && !istype(G, /obj/item/clothing/gloves/color/yellow/sprayon)) //sorry for snowflake
```
It seems like this would cause disparity between these and insuls, is this the case/intended?

yeeyeh replied:
So essentially what the !istype is doing is executing electrocute_act anyway if the gloves are spray-ons, since otherwise it never calls electrocute_act and thus never decrements shocks_remaining, leading to the gloves simply acting like non-removable insuls, instead of limited use ones.

Qustinnus replied:
this is way too snowflake

LemonInTheDark reviewed (commented):
Based. just a few nitpicks on my part

fludd12 commented:
Malt AI loss condition

ArcaneMusic commented:
> I hate it and love it at the same time.

This reaction means you have made a pretty much on-brand minor item.

carshalash commented:
I get this reference.

trollbreeder commented:
I get this reference too.
What's next, spray on galoshes?

Shadowflame909 commented:
Remember you'd be spending 10k for insuls when you can by death squad 0.5 armor and syndicate gloves for 3k.

This is based

mrhugo13 commented:
Now make a discount version which just paints the users hands yellow and doesn't actually do anything besides make them look stupid.

PKPenguin321 commented on code/modules/clothing/gloves/color.dm:
```diff
@@ -12,6 +12,41 @@
+
+/obj/item/clothing/gloves/color/yellow/sprayon/Initialize()
+	.=..()
+	ADD_TRAIT(src, TRAIT_NODROP, CURSED_ITEM_TRAIT) //Cursed with the power of science, or something
```
I dislike this being an actual item that blocks other gloves, would make more sense as a trait or a component that grants a trait

trollbreeder replied:
but that'd destroy the reference pkpenguin

itseasytosee replied:
wait so you could cut off your arm and get an actual pair of gloves that someone else could but on?

yeeyeh replied:
For PKPenguin, it's intended behavior to make it block other gloves. They're supposed to be unremovable and unreplaceable as according to their reference. If you really wanted to, it would be pretty simple to just shock yourself 10 times for them to break.
And for itseasytosee, I'm just gonna chalk it up to SS13 logic, since you also need two arms to wear a gloves at all.

PKPenguin321 replied:
I guess it works for the reference. They should delete when dropped though since generally you don't want NODROP items lying around

yeeyeh replied:
i see where you're coming from, fixed

itseasytosee commented:
What is the number one problem facing our community today? Ungloved greytiders. which I why I've invented an easy alternative hand covering. Spray on gloves. Voila. 
How ya gonna get em off nerd?
Hahahaha
What a freak.
He wants to be smart but that lame.
I wanted to run away that day, but you can't run away from your own gloves.

ATH1909 commented:
How much is 10 shocks' worth of protection? It'll protect you from running into a shocked door, sure, but IIRC, you'll burn through all 10 of those shocks very quickly if you accidentally shock a door while you have its hacking interface open.

yeeyeh commented:
While I was testing, 10 shocks took me spamming an electrified wire to make it go by quickly, and I believe that if you stand diagonal to the door the sparks flying out of it won't touch you. If you already know the test wire you could get through 10 doors, essentially.

yeeyeh committed:
gloves go bye bye if you lose an arm

yeeyeh committed:
whoops

SkeletalElite commented:
> While I was testing, 10 shocks took me spamming an electrified wire to make it go by quickly, and I believe that if you stand diagonal to the door the sparks flying out of it won't touch you. If you already know the test wire you could get through 10 doors, essentially.

It doesn't matter where you stand. The door only shocks you when you interact with it. IE attempt to open it or pulse / cut a wire while it is electrified.

nemvar commented on code/modules/clothing/gloves/color.dm:
```diff
@@ -12,6 +12,45 @@
+
+/obj/item/clothing/gloves/color/yellow/sprayon/dropped()
+	.=..()
+	qdel(src) //loose nodrop items bad
```
There is a DROP_DEL flag in item_flags. Use that instead.

yeeyeh replied:
dropped() is better than dropdel for this because dropdel only deletes if it hits the floor, instead of being moved from its slot at all

nemvar commented on code/modules/mob/living/carbon/human/human_defense.dm:
```diff
@@ -463,6 +463,10 @@
 /mob/living/carbon/human/electrocute_act(shock_damage, source, siemens_coeff = 1, flags = NONE)
+	if(istype(gloves, /obj/item/clothing/gloves/color/yellow/sprayon)) //Checks to see if the gloves are spray-on, if yes, damage the gloves.
+		var/obj/item/clothing/gloves/color/yellow/sprayon/temp = gloves
+		temp.Shocked()
```
Listen to the COMSIG_MOB_ELECTROCUTE signal instead.

nemvar commented on code/modules/clothing/gloves/color.dm:
```diff
@@ -12,6 +12,45 @@
+			C.visible_message("<span class='notice'>[U] sprays their hands with glittery rubber!</span>")
+		else
+			C.visible_message("<span class='warning'>[U] sprays glittery rubber on the hands of [C]!</span>")
+		C.equip_to_slot_or_del(new /obj/item/clothing/gloves/color/yellow/sprayon,ITEM_SLOT_GLOVES) //we're gonna pretend that the rubber can't stick to gloves.
```
Some kind of feedback if this fails would probably be nice.

yeeyeh committed:
merge conflicts? in MY pr? its more likely than you think

now featuring Signals™

yeeyeh committed:
Merge branch 'master' into insuls-in-a-can

yeeyeh committed:
travis you fuck

also adds feedback if you cant spray them on

yeeyeh committed:
Merge remote-tracking branch 'origin/insuls-in-a-can' into insuls-in-a-can

yeeyeh committed:
forgot to remove a commented out line

yeeyeh committed:
Merge branch 'master' into insuls-in-a-can

AnturK commented on code/modules/clothing/gloves/color.dm:
```diff
@@ -12,6 +12,52 @@
+
+/obj/item/clothing/gloves/color/yellow/sprayon/Initialize()
+	.=..()
+	ADD_TRAIT(src, TRAIT_NODROP, CURSED_ITEM_TRAIT) //Cursed with the power of science, or something
```
It's not cursed item, change the source to something more appriopriate.

Iamgoofball replied:
To be fair it's pretty cursed but not in that way

AnturK commented on code/modules/clothing/gloves/color.dm:
```diff
@@ -12,6 +12,52 @@
+
+/obj/item/clothing/gloves/color/yellow/sprayon/equipped(mob/user, slot)
+	. = ..()
+	RegisterSignal(user, COMSIG_LIVING_ELECTROCUTE_ACT, .proc/Shocked)
```
This signal doesn't seem to actually matter for damage received or anything ?

yeeyeh replied:
https://discordapp.com/channels/326822144233439242/326831214667235328/675423475610943491
I'm pretty sure this is in reference to this request, and it's also outdated anyway. Reopen if I'm wrong.

AnturK commented on code/modules/power/power.dm:
```diff
@@ -291,7 +291,7 @@
 		if(H.gloves)
 			var/obj/item/clothing/gloves/G = H.gloves
-			if(G.siemens_coefficient == 0)
+			if(G.siemens_coefficient == 0 && !istype(G, /obj/item/clothing/gloves/color/yellow/sprayon)) //sorry for snowflake
```
This is pretty ugly.

yeeyeh committed:
Removes snowflake

that moment when you lose all motivation to code after getting a review so you put it off for a week and then it goes stale 5 minutes after you start working on it again

yeeyeh committed:
Merge remote-tracking branch 'origin/insuls-in-a-can' into insuls-in-a-can

yeeyeh committed:
Uncurses gloves

tralezab reviewed (approved):
code looks good 😋

ArcaneMusic commented on code/modules/clothing/gloves/color.dm:
```diff
@@ -12,6 +12,52 @@
 	custom_premium_price = 1200
 
+/obj/item/toy/sprayoncan
+	name = "spray-on insuls applicator"
```
Might want to go with a formal name than this one, like just "spray-on insulation applicator".

Iamgoofball replied:
Seeing as this is venture capitalism the machine, something a bit more snappy and marketable would probably be better. Think those god awful Billy Mays ads.

tralezab commented:
excuse me i didn't mean to do that three times oh well you should merge this 😨

yeeyeh committed:
formalities

thank you arcane man

actioninja commented on code/modules/research/bepis.dm:
```diff
@@ -36,7 +36,8 @@
 					/obj/item/pen/survival,
-					/obj/item/circuitboard/machine/sleeper/party)
+					/obj/item/circuitboard/machine/sleeper/party,
+          /obj/item/toy/sprayoncan)
```
tabs are used in this file, change the spaces here

yeeyeh committed:
formatting fix

i swear i used the tab button but i guess i am just cursed

bobbahbrown commented on code/modules/clothing/gloves/color.dm:
```diff
@@ -12,6 +12,52 @@
+			else
+				C.visible_message("<span class='warning'>[U] sprays glittery rubber on the hands of [C]!</span>")
+		else
+			C.visible_message("<span class='warning'>The rubber fails to stick to [C]'s hands!")
```
```suggestion
			C.visible_message("<span class='warning'>The rubber fails to stick to [C]'s hands!</span>")

```

bobbahbrown replied:
missing closing tag

yeeyeh committed:
thank you strange bee creature

thank you strange bee creature

Co-Authored-By: Bobbahbrown <bobbahbrown@gmail.com>
